trl/trainer/grpo_trainer.py

```diff
@@ -244,8 +244,24 @@ def __init__(
                 )
 
         if peft_config is not None:
+            if not is_peft_available():
+                raise ImportError("PEFT is required to use `peft_config`. Run `pip install peft`.")
             model = get_peft_model(model, peft_config)
 
+        # Enable gradient checkpointing if requested
+        if args.gradient_checkpointing:
+            # Ensure use_cache is disabled
+            if hasattr(model, "config"):
+                model.config.use_cache = False
```

Comment:
```suggestion
             model.config.use_cache = False
```

is should always be the case